src/libstore/globals.cc

```diff
@@ -143,15 +143,19 @@ std::vector<Path> getUserConfigFiles()
     return files;
 }
 
+unsigned int Settings::getDefaultThreads()
+{
+    return std::max(1U, std::thread::hardware_concurrency());
```

Comment:
I was wrong earlier today to point out that we should factor out the hardware_concurrency part only. `getMaxCPU` is equally relevant to jobs, I'd say, so let's reuse the whole `getDefaultCores()`.

The name of that function is too suggestive of its use though (only the `"cores"` setting), so let's rename that one to something like `getAllowedCoreCount()`.

Reply:
My goal with creating this function was just to have one source of truth for the `auto` value for `max-jobs`.

I've done further refactoring to make this clearer 👍